src/NServiceBus.Core/Pipeline/Incoming/ReceivePipelineConfigExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    /// <summary>
+    /// Provides access to receive pipeline notifications.
+    /// </summary>
+    public static class ReceivePipelineConfigExtensions
+    {
+        /// <summary>
+        /// Subscribes to notifications for completed receive pipelines.
+        /// </summary>
+        public static void OnReceivePipelineCompleted(this PipelineSettings pipelineSettings, Func<ReceivePipelineCompleted, Task> subscription)
```

Comment:
Synced with @timbussmann and decided to go ahead with this but to align the error notifications in a separate PR